client/my-sites/email/email-management/home/email-no-domain.jsx

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import { isFreePlan } from '@automattic/calypso-products';
+import { localize } from 'i18n-calypso';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import EmptyContent from 'calypso/components/empty-content';
+import Illustration from 'calypso/assets/images/customer-home/illustration--task-find-domain.svg';
+
+const EmailNoDomain = ( { selectedSite, translate } ) => {
+	if ( isFreePlan( selectedSite.plan.product_slug ) ) {
+		return (
+			<EmptyContent
+				action={ translate( 'Upgrade', { context: 'verb' } ) }
+				actionURL={ `/plans/${ selectedSite.slug }` }
+				illustration={ Illustration }
+				line={ translate(
+					'Upgrade now, claim your domain and pick from one of our flexible options to connect your domain with email and start getting emails today.'
+				) }
+				title={ translate( 'Get your own domain for a custom email address' ) }
+			/>
+		);
+	}
+
+	return (
+		<EmptyContent
+			action={ translate( 'Add a Domain' ) }
+			actionURL={ `/domains/add/${ selectedSite.slug }` }
+			illustration={ Illustration }
+			line={ translate(
+				'Claim your domain, pick from one of our flexible options to connect your domain with email and start getting emails today.'
+			) }
+			title={ translate( 'Claim your free domain to use with a custom email address' ) }
```

Comment:
Thank you again for the ping, @daledupreez! 

>  I think we should only show this content if we know the user has the domain credit. 

+1 to this! Yes, we should only show this if the user has a domain that they can claim. Otherwise that would be very frustrating for the user.

> Unfortunately, that may require us to wire in the hasDomainCredit selector.
How much of a trouble is this? To me it's better to have certainty in our messaging tone, rather than say "this or this".

Just to clarify what would the CTA would say if this is the case? When exactly in this flow we'll know user is claiming v.s. purchasing? And how is that communicated to the user? Sorry for too many questions!
@semihakocer what are your thoughts?

Reply:
I plan to deploy this change and then take on the improvements in a follow-up PR. In particular, I think we need to do the following:

### Site is on the free plan

**Get your own domain for a custom email address**
Upgrade now, ~claim~ set up your domain and pick from one of our flexible options to connect your domain with email and start getting emails today.

### Site is on a paid plan with unused domain credit

**Claim your free domain to use with a custom email address**
Claim your domain, pick from one of our flexible options to connect your domain with email and start getting emails today.

### Site is on a paid plan but doesn't have a domain credit

**Set up a domain to use with a custom email address**
Set up or buy your domain, pick from one of our flexible email options, and start getting emails today.